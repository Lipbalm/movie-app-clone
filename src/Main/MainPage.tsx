import React, { FC } from "react";
import { CarouselContainer, BodyWrapper } from "../Styles/MainStyle";
import CardWidget from "../Components/WidgetComponents/CardWidget";
import { tempCards } from "../Test/TempData/Card";

const MainPage: FC = () => {
  return (
    <>
      <CarouselContainer></CarouselContainer>
      <BodyWrapper>
        <CardWidget
          cards={tempCards}
          head_text="2020년 1분기 기대신작 애니"
          key="card_widget_1"
        ></CardWidget>
      </BodyWrapper>
    </>
  );
};

export default MainPage;
