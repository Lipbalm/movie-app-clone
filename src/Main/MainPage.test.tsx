import React from "react";
import { render, fireEvent } from "@testing-library/react";
import MainPage from "./MainPage";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";

function setup() {
  const history = createMemoryHistory();
  return {
    ...render(
      <Router history={history}>
        <MainPage />
      </Router>
    )
  };
}

describe("<MainPage />", () => {
  it("snapshot", () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });
});
