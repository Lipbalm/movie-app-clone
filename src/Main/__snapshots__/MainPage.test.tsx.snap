// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainPage /> snapshot 1`] = `
<div>
  <div
    class="sc-bxivhb czzmdl"
  />
  <div
    class="sc-ifAKCX cktARQ"
  >
    <div
      class="sc-EHOje hjUbwC"
    >
      <h2
        class="sc-bZQynM kXqyJF"
      >
        제일 많이 본 영화
        <a
          class="sc-gzVnrw ddfmCq"
          href="/item"
        >
          더보기
        </a>
      </h2>
      <div
        class="sc-kGXeez jrrZSC"
      >
        <div
          class="sc-kpOJdX cpJozj"
        />
      </div>
      <div
        class="sc-htoDjs ilhYAR"
      >
        <div
          class="sc-dnqmqq fMKVSZ"
        >
          <div
            class="sc-gZMcBi Uwzts"
            data-testid="card"
          >
            <div
              class="sc-VigVT knNAAJ"
            >
              <div
                class="sc-jTzLTM dodnuM"
              />
            </div>
            <div
              class="sc-jzJRlG dpMOwe"
            >
              <span
                class="sc-kAzzGY ggQPml"
                data-testid="genre"
              >
                35/18/10749
              </span>
              <a
                class="sc-chPdSV ijoXjQ"
                data-testid="title"
              >
                Dilwale Dulhania Le Jayenge
              </a>
            </div>
          </div>
          <div
            class="sc-gZMcBi Uwzts"
            data-testid="card"
          >
            <div
              class="sc-VigVT knNAAJ"
            >
              <div
                class="sc-jTzLTM koSzvs"
              />
            </div>
            <div
              class="sc-jzJRlG dpMOwe"
            >
              <span
                class="sc-kAzzGY ggQPml"
                data-testid="genre"
              >
                80/18
              </span>
              <a
                class="sc-chPdSV ijoXjQ"
                data-testid="title"
              >
                The Shawshank Redemption
              </a>
            </div>
          </div>
          <div
            class="sc-gZMcBi Uwzts"
            data-testid="card"
          >
            <div
              class="sc-VigVT knNAAJ"
            >
              <div
                class="sc-jTzLTM cTdkRs"
              />
            </div>
            <div
              class="sc-jzJRlG dpMOwe"
            >
              <span
                class="sc-kAzzGY ggQPml"
                data-testid="genre"
              >
                18/80
              </span>
              <a
                class="sc-chPdSV ijoXjQ"
                data-testid="title"
              >
                The Godfather
              </a>
            </div>
          </div>
          <div
            class="sc-gZMcBi Uwzts"
            data-testid="card"
          >
            <div
              class="sc-VigVT knNAAJ"
            >
              <div
                class="sc-jTzLTM fIpddy"
              />
            </div>
            <div
              class="sc-jzJRlG dpMOwe"
            >
              <span
                class="sc-kAzzGY ggQPml"
                data-testid="genre"
              >
                18/36/10752
              </span>
              <a
                class="sc-chPdSV ijoXjQ"
                data-testid="title"
              >
                Schindler's List
              </a>
            </div>
          </div>
          <div
            class="sc-gZMcBi Uwzts"
            data-testid="card"
          >
            <div
              class="sc-VigVT knNAAJ"
            >
              <div
                class="sc-jTzLTM fZxKFo"
              />
            </div>
            <div
              class="sc-jzJRlG dpMOwe"
            >
              <span
                class="sc-kAzzGY ggQPml"
                data-testid="genre"
              >
                35/18/53
              </span>
              <a
                class="sc-chPdSV ijoXjQ"
                data-testid="title"
              >
                Parasite
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="sc-EHOje hjUbwC"
    >
      <h2
        class="sc-bZQynM kXqyJF"
      >
        제일 많이 본 영화
        <a
          class="sc-gzVnrw ddfmCq"
          href="/itme"
        >
          더보기
        </a>
      </h2>
      <div
        class="sc-kGXeez jrrZSC"
      >
        <div
          class="sc-kpOJdX cpJozj"
        />
      </div>
      <div
        class="sc-htoDjs ilhYAR"
      >
        <div
          class="sc-iwsKbI cwFckA"
        >
          <div
            class="sc-gqjmRU ddZkBY"
            data-testid="item"
          >
            <div
              class="sc-VigVT fiDZiM"
              data-testid="thumnailWrapper"
            >
              <div
                class="sc-jTzLTM gWqGFT"
                data-testid="thumnail"
              />
            </div>
            <div
              class="sc-fjdhpX cCSzsK"
              data-testid="layer"
            >
              <a
                class="sc-kgoBCf hNMDMm"
                data-testid="link"
              >
                Dilwale Dulhania Le Jayenge
              </a>
            </div>
          </div>
        </div>
        <div
          class="sc-iwsKbI hloxKn"
        >
          <div
            class="sc-gqjmRU ddZkBY"
            data-testid="item"
          >
            <div
              class="sc-VigVT hlRwKS"
              data-testid="thumnailWrapper"
            >
              <div
                class="sc-jTzLTM juYtdO"
                data-testid="thumnail"
              />
            </div>
            <div
              class="sc-fjdhpX cCSzsK"
              data-testid="layer"
            >
              <a
                class="sc-kgoBCf hNMDMm"
                data-testid="link"
              >
                Dilwale Dulhania Le Jayenge
              </a>
            </div>
          </div>
          <div
            class="sc-gqjmRU bJQSHF"
            data-testid="item"
          >
            <div
              class="sc-VigVT hlRwKS"
              data-testid="thumnailWrapper"
            >
              <div
                class="sc-jTzLTM hWcngc"
                data-testid="thumnail"
              />
            </div>
            <div
              class="sc-fjdhpX cCSzsK"
              data-testid="layer"
            >
              <a
                class="sc-kgoBCf hNMDMm"
                data-testid="link"
              >
                The Shawshank Redemption
              </a>
            </div>
          </div>
          <div
            class="sc-gqjmRU bJQSHF"
            data-testid="item"
          >
            <div
              class="sc-VigVT hlRwKS"
              data-testid="thumnailWrapper"
            >
              <div
                class="sc-jTzLTM bGTElw"
                data-testid="thumnail"
              />
            </div>
            <div
              class="sc-fjdhpX cCSzsK"
              data-testid="layer"
            >
              <a
                class="sc-kgoBCf hNMDMm"
                data-testid="link"
              >
                The Godfather
              </a>
            </div>
          </div>
          <div
            class="sc-gqjmRU bJQSHF"
            data-testid="item"
          >
            <div
              class="sc-VigVT hlRwKS"
              data-testid="thumnailWrapper"
            >
              <div
                class="sc-jTzLTM djSytx"
                data-testid="thumnail"
              />
            </div>
            <div
              class="sc-fjdhpX cCSzsK"
              data-testid="layer"
            >
              <a
                class="sc-kgoBCf hNMDMm"
                data-testid="link"
              >
                Schindler's List
              </a>
            </div>
          </div>
          <div
            class="sc-gqjmRU cjbSoN"
            data-testid="item"
          >
            <div
              class="sc-VigVT hlRwKS"
              data-testid="thumnailWrapper"
            >
              <div
                class="sc-jTzLTM hdRkPf"
                data-testid="thumnail"
              />
            </div>
            <div
              class="sc-fjdhpX cCSzsK"
              data-testid="layer"
            >
              <a
                class="sc-kgoBCf hNMDMm"
                data-testid="link"
              >
                Parasite
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
